core/opendaq/signal/include/opendaq/signal_impl.h

```diff
@@ -436,7 +438,12 @@ ErrCode SignalBase<TInterface, Interfaces...>::sendPacket(IPacket* packet)
     std::scoped_lock lock(this->sync);
 
     if (sendPacketInternal(packetPtr))
+    {
+        const auto dataPacket = packetPtr.asPtrOrNull<IDataPacket>();
+        if (dataPacket.assigned() && dataPacket.getSampleCount())
+            lastDataPacket = dataPacket;
```

Comment:
I'd add an option to disable this feature on a private interface. If disabled, `getValue` should either fail or return nullptr. Devices might sometimes have their own packet pools that could contain only a small amount of packets. In that case always holding onto a packet might be a significant overhead.